ci-operator/step-registry/cucushift/upgrade/eus/unpause-worker-mcp/cucushift-upgrade-eus-unpause-worker-mcp-commands.sh

```diff
@@ -19,32 +19,40 @@ function debug() {
     fi
 }
 
+# unpause the paused mcp
 function unpause() {
-    oc patch --type=merge --patch='{"spec":{"paused":false}}' machineconfigpool/worker
-    ret=$(oc get machineconfigpools worker -ojson| jq .spec.paused)
+    echo "Unpause the paused mcp..."
+    local mcp="$1"
+    oc patch --type=merge --patch='{"spec":{"paused":false}}' mcp/${mcp}
+    if [ $? -ne 0 ]; then
+        echo "Unpause mcp failed"
+        exit 1
+    fi
+
+    ret=$(oc get machineconfigpools ${mcp} -ojson | jq -r '.spec.paused')
     if [[ "${ret}" != "false" ]]; then
-        echo >&2 "Failed to resume worker pool, exiting..." && return 1
+        echo >&2 "${mcp} pool failed to unpause, exiting" && exit 1
     fi
 }
 
 function check_mcp() {
-    local out updated updating degraded try=0 max_retries=30
+    local out updated try=0 max_retries=30
+    local mcp="$1" expected_status="$2"
+    echo "Checking mcp ${mcp}, expected status ${expected_status}..."
     while (( try < max_retries )); 
     do
+        sleep 300
         echo "Checking worker pool status #${try}..."
-        run_command "oc get machineconfigpools"
-        out="$(oc get machineconfigpools worker --no-headers)"
+        out="$(oc get machineconfigpools ${mcp} --no-headers)"
+        echo $out
         updated="$(echo "${out}" | awk '{print $3}')"
-        updating="$(echo "${out}" | awk '{print $4}')"
-        degraded="$(echo "${out}" | awk '{print $5}')"
     
-        if [[ ${updated} == "True" && ${updating} == "False" && ${degraded} == "False" ]]; then
```

Comment:
shall we keep the check on `updating` and `degraded` in check_mcp?

Reply:
done